File: my-solutions/0209-minimum-size-subarray-sum/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0, r = 0;
        int n = nums.size();
        long sum = 0;
        int minLength = INT_MAX;

        while (r < n) {
            sum += nums[r];
            while (sum >= target) {
                minLength = min(minLength, (r - l + 1));
                if (minLength == 1) {return 1;}
                sum -= nums[l];
                l++;
            }
            r++;
        }

        return (minLength == INT_MAX) ? 0 : minLength;
    }
};

File: my-solutions/0209-minimum-size-subarray-sum/solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start, end, running = 0, 0, 0
        minWindow = float('inf')

        while end < len(nums):
            running += nums[end]
            while start <= end and running >= target:
                minWindow = min(minWindow, end - start + 1)
                if start < end:
                    running -= nums[start]
                    start += 1
                else:
                    break
            end += 1
        
        return minWindow if minWindow != float('inf') else 0
